v0/pipeline/steps/cd/k8s-traffic-routing-rule.yaml

```diff
@@ -0,0 +1,236 @@
+title: TrafficRouteRule
+type: object
+required:
+  - type
+  - spec
+properties:
+  type:
+    type: string
+    enum:
+      - uri
+      - method
+      - header
```

Comment:
Not all of these enums apply for all route types. Can you check how can we be more specific?

Reply:
I was looking into it and didn't find an easy way of doing it through static schema as there is no way of passing some enum value from parent to child schema. To do that we would need to create a specific schema for each Provider - Route - Rule relation. For example Schema for SmiHttpUri, IstioHttpUri, SmiHttpPort, IstioHttpPort... and that would increase complexity a lot. I found this implementation optimal and the idea is to do additional validation on the backend when the time comes.